finality-aleph/src/network/manager/service.rs

```diff
@@ -134,30 +150,35 @@ impl<NI: NetworkIdentity, D: Data> Service<NI, D> {
         }
     }
 
-    fn delete_reserved(to_remove: HashSet<PeerId>) -> Option<ConnectionCommand> {
+    fn delete_reserved(
+        to_remove: HashSet<NI::PeerId>,
+    ) -> Option<ConnectionCommand<NI::Multiaddress>> {
         match to_remove.is_empty() {
             true => None,
             false => Some(ConnectionCommand::DelReserved(to_remove)),
         }
     }
 
-    fn finish_session(&mut self, session_id: SessionId) -> Option<ConnectionCommand> {
+    fn finish_session(
+        &mut self,
+        session_id: SessionId,
+    ) -> Option<ConnectionCommand<NI::Multiaddress>> {
         self.sessions.remove(&session_id);
         self.to_retry
             .retain(|(pre_session, _)| pre_session.session_id() != session_id);
         Self::delete_reserved(self.connections.remove_session(session_id))
     }
 
     fn network_message(
-        (message, command): (DiscoveryMessage, DataCommand),
-    ) -> (NetworkData<D>, DataCommand) {
+        (message, command): (DiscoveryMessage<NI::Multiaddress>, DataCommand<NI::PeerId>),
```

Comment:
At the risk of sounding like a broken record: structs here would make it more future-proof as you could change the field order :sunglasses: